modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -1190,6 +1211,16 @@ private CompletableFuture<Void> handleChangePendingAssignmentEvent(Entry pending
                     return nullCompletedFuture();
                 }
 
+                // A node might exist in stable yet we don't want to start the replica
```

Comment:
Why do we have the same code here and in `handleChangePendingAssignmentEvent`?

Reply:
In table manager we have only one place where we use logic like this: 
```
                    if (isRecovery && !replicaMgr.isReplicaStarted(replicaGrpId)) {
                        return;
                    }
```
Why in colocation logic we have two places? What is the reason?